### PR TITLE
Change default perl5 to 5.34

### PR DESCRIPTION
**Pull request by jmroot**
Older perl versions are EOL: https://www.cpan.org/src/README.html

Also removed perl5.26 variants that can't work because the dependencies no longer exist.

### CODE REVIEW

**Comment by i0ntempest** [comment]
Awesome, this is finally being done

**Comment by nortcele** [comment]
It's fine for me.

**Review by xeron — APPROVED** [review]

**Review by i0ntempest — APPROVED** [review]

**Review by mascguy — APPROVED** [review]
Works for me!

**Comment by drkp** [comment]
LGTM for my ports too. Thanks!

**Review by blair — APPROVED** [review]

File: audio/flac2mp3/Portfile [hunk 1]
```diff
@@ -30,11 +29,10 @@ checksums           rmd160  eb5d32d163f217ccbd5c2c969159486adcf922be \
 distname            ${name}
 
 # add perl variants
-# set default_branch to 5.28
 perl5.require_variant   yes
 perl5.conflict_variants yes
-perl5.branches          5.26 5.28 5.30
-perl5.default_branch    5.28
+perl5.branches          5.28 5.30 5.32 5.34
```
Comment (mojca) [inline audio/flac2mp3/Portfile:34]: Honestly I don't see much point in adding variants for 5.32 etc., but it doesn't really hurt either.

**Comment by mojca** [comment]
@jmroot: awesome, thank you very much for the nice work you did. Now I still see the following ports depending on perl 5.28:
```
PsyncX	ryandesign,openmaintainer
amttools	Ionic,openmaintainer
conf2struct	amake,openmaintainer
ddclient	nerdling,openmaintainer
evolution-data-server	mascguy,openmaintainer
flamegraph	
frozenbubble2	ryandesign,openmaintainer
gnome-voice	oleaamot,openmaintainer
gscan2pdf	
lilypond	nerdling,openmaintainer
lilypond-devel	nerdling,openmaintainer
macos-fortress-dshield	essandess,openmaintainer
macos-fortress-hosts	essandess,openmaintainer
myrepos	gjasny_googlemail.com,openmaintainer
mythtv-core.27	ctreleaven,openmaintainer
mythtv-plugins.27	ctreleaven,openmaintainer
mythweb.27	ctreleaven,openmaintainer
mythweb.28	ctreleaven,openmaintainer
pulseaudio	Ionic,openmaintainer
samba4	
stow	
swaks	
vislcg3	consult@tinodidriksen.com TinoDidriksen
```
as generated by
```
port info --line --index --name --maintainer depends:'perl5\.28' or depends:'p5\.28-' | grep -v 'p5-' | grep -v 'p5.28-' | sort
```
and potentially biased by my locally installed ports. There's a bunch more that depend on 5.30, maybe also on 5.32.

My preference would be to completely remove support for anything older than 5.34, so that we can remove all the thousands of subports.

**Comment by jmroot** [comment]
> @jmroot: Now I still see the following ports depending on perl 5.28:

Some of these I didn't update because they have bugs with perl version selection, like not using a fixed version (which is of course incompatible with binary distribution). The idea was to address those separately. Others I may have legitimately missed.

**Pull request merged**